src/data/bucket/line_bucket.js

```diff
@@ -110,20 +124,78 @@ class LineBucket implements Bucket {
         this.layers = options.layers;
         this.layerIds = this.layers.map(layer => layer.id);
         this.index = options.index;
+        this.features = [];
 
         this.layoutVertexArray = new LineLayoutArray();
         this.indexArray = new TriangleIndexArray();
         this.programConfigurations = new ProgramConfigurationSet(layoutAttributes, options.layers, options.zoom);
         this.segments = new SegmentVector();
+
+        this.dataDrivenPatternLayers = [];
+
+
+        for (let i = 0; i < this.layerIds.length; i++) {
+            const id = this.layerIds[i];
+            const programConfiguration = this.programConfigurations.get(id);
+            if (programConfiguration.binders && programConfiguration.binders['line-pattern'] && programConfiguration.binders['line-pattern'].isDataDriven()) {
+                this.dataDrivenPatternLayers.push(i);
+            }
+        }
```

Comment:
Rather than storing this on `this`, could we compute it locally in `populate()`?   (Also: why is this an array of indices into `this.layers` rather than an array of layers themselves?)

Reply:
One more question: why use `binder.isDataDriven()` rather than checking `layer.paint.get('line-pattern')`?